packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -232,22 +232,36 @@ export interface EcsOptimizedAmiProps {
   /**
    * What generation of Amazon Linux to use
    *
-   * @default AmazonLinux
+   * @deprecated Use amiType instead.
    */
   readonly generation?: ec2.AmazonLinuxGeneration;
+
+  /**
+   * What ECS Optimized AMI type to use
+   *
+   * @default is Amazon Linux
+   */
+  readonly amiType?: EcsOptimizedAmiType;
 }
 
 /**
  * Construct a Linux machine image from the latest ECS Optimized AMI published in SSM
  */
 export class EcsOptimizedAmi implements ec2.IMachineImageSource {
   private readonly generation: ec2.AmazonLinuxGeneration;
+  private readonly amiType: EcsOptimizedAmiType;
+
   private readonly amiParameterName: string;
 
   constructor(props?: EcsOptimizedAmiProps) {
     this.generation = (props && props.generation) || ec2.AmazonLinuxGeneration.AmazonLinux;
-    if (this.generation === ec2.AmazonLinuxGeneration.AmazonLinux2) {
+    this.amiType = (props && props.amiType) || EcsOptimizedAmiType.AmazonLinux;
```

Comment:
Please make it disallowed to supply both `amiType` and `generation`.

Also, are we 100% sure that we'll never offer any other OS options than amazon linux 2 that offer a GPU and/or ARM version? Because if we are, they should be 2 different axes: Distro + Hardware.

Reply:
How about if we leave the `generation` and change `amiType` to `hwType`? That way you can specify no input parameter and it will default to `AmazonLinux` if hardware is `standard` and defaults to `AmazonLinux2` if hardware is `ARM` or `GPU`. Will throw error if `generation` is defined to be `AmazonLinux` and hardware is `GPU` or `ARM`.

 Allows for 2 axis of Distro + HW for future options (e.g. Ubuntu or others).

Reply:
That's what I was thinking, yes.

Reply:
ok, I will change to have `generation` and `hw` params in the `EcsOptimizedAmiProps` class. will also add a couple of unit tests